### ISSUE TITLE
helm: support creating `cilium-ipsec-keys` secret

### ISSUE THREAD

**Issue by austince** [cilium/cilium#18084]
## Proposal / RFE

**Is your feature request related to a problem?**

Nope.

**Describe the solution you'd like**

Support for creating the `cilium-ipsec-keys` secret (and equivalent one for wireguard) via the helm chart. Currently, this must be created separately, see: https://docs.cilium.io/en/v1.10/gettingstarted/encryption-ipsec/#generate-import-the-psk

**Comment by pchaigno** [comment]
Did you consider using the Cilium CLI? It can create the IPsec secret for you.

**Comment by austince** [comment]
Yes, I did consider it but that adds additional tooling to build workflows, etc. around. If using Helm, it would be nice to not have to use any other tool.

**Comment by aanm** [comment]
@austince do you want to submit a PR? I'm not sure if helm supports such feature but if it does it might be easy to write a solution for this issue.

**Comment by austince** [comment]
Sure – I think its sufficient to just create the Secret resource given a value, and not worry about generating the value in Helm.

**Comment by j-land** [comment]
Status on this?

**Comment by j-land** [comment]
I don't at the moment :(

**Comment by chaunceyjiang** [comment]
Hi @austince 

Do you have free time to complete this issue? if not, i can continue to work on this based on your PR.

**Comment by austince** [comment]
I don't, unfortunately -- would be great to see you land that though!
